pkg/webhook/immutable_fields_validator.go

```diff
@@ -294,6 +294,9 @@ func validateImmutableFieldsForTFBasedResource(obj, oldObj *unstructured.Unstruc
 		return admission.Errored(http.StatusBadRequest,
 			fmt.Errorf("couldn't get ResourceConfig for kind %v: %w", obj.GetKind(), err))
 	}
+	if rc.Direct && rc.Name != "google_sql_database_instance" {
```

Comment:
cc @jasonvigil  Just FYI, we skip the immutable webhook check for other direct resources, and leave the sql there to make sure its behavior is unchanged.